src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -725,7 +725,7 @@ public static IGraphQLBuilder UseMiddleware<TMiddleware>(this IGraphQLBuilder bu
             if (serviceLifetime == ServiceLifetime.Scoped)
             {
                 // this code prevents registrations of scoped middleware for a singleton schema, which is impossible.
-                throw new ArgumentOutOfRangeException("Please specify a transient or singleton service lifetime. Specifying transient will cause the middleware lifetime to match that of the schema. Using a scoped schema will then have scoped middleware.");
+                throw new ArgumentOutOfRangeException(nameof(serviceLifetime), "Please specify a transient or singleton service lifetime. Specifying transient will cause the middleware lifetime to match that of the schema. Using a scoped schema will then have scoped middleware.");
```

Comment:
"Please specify a ..." was treated as argument name.

Reply:
Good catch

Reply:
By Roslyn, not me.